### ISSUE TITLE
Support Python 3.13

### ISSUE THREAD

**Issue by musicinmybrain** [pydantic/pydantic#9668]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

Please consider supporting Python 3.13 soon – or at least commenting on a timeline for doing so.

- Python 3.13 is now at [3.13.0b2](https://www.python.org/downloads/release/python-3130b2/), with [release candidate 1 scheduled for the end of July and the final release scheduled for the beginning of October](https://peps.python.org/pep-0719/).
- Fedora now has [Python 3.13 as the system Python](https://fedoraproject.org/wiki/Changes/Python3.13) in Rawhide, and [Fedora 41 will release with Python 3.13 in October](https://fedorapeople.org/groups/schedule/f-41/f-41-all-tasks.html).
- [PyO3 is planning to release 0.22 soon](https://github.com/PyO3/pyo3/issues/4038#issuecomment-2156400881) with “official” Python 3.13 support.

-----

In Fedora, we’ve managed to rebuild the `python-pydantic-core` package with Python 3.13, and all of its tests pass.

However:

- Test collection errors due to the `DeprecationWarning` from https://github.com/pydantic/pydantic/issues/9613.
- With `pytest -W ignore::DeprecationWarning`, test collection errors due to `TypeError: typing.Annotated cannot be used with issubclass()`, similar to the previous reports https://github.com/pydantic/pydantic/issues/8534 and https://github.com/pydantic/pydantic/issues/7942.
- With `python -m pytest -W 'ignore::DeprecationWarning' -v --ignore=tests/test_type_adapter.py`, test collection succeeds, but there are many errors. A large portion appear to be the same `TypeError: typing.Annotated cannot be used with issubclass()`, but it appears there are also some unrelated assertion failures.

It’s currently possible to test Pydantic in a Python 3.13 virtualenv with:

```
$ gh repo clone pydantic/pydantic
$ cd pydantic
$ python3.13 -m venv _e
$ . _e/bin/activate
(_e) $ UNSAFE_PYO3_SKIP_VERSION_CHECK=1 pip install -e .
(_e) $ pip install cloudpickle coverage[toml] dirty-equals eval-type-backport pytest pytest-mock pytest-pretty pytest-examples faker pytest-benchmark pytest-codspeed packaging
(_e) $ python -m pytest
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
@musicinmybrain,

Thanks for opening this feature request. Support for 3.13 is definitely something we want to do explicitly soon. I'll chat with the team re timeline and update here afterwards.

**Comment by sydney-runkle** [comment]
Definitely something we're prioritizing.

I'll pull this in once we get the new version of `pyo3` into `pydantic-core` with 3.13 support. Hoping to have this included in v2.8 around the end of the month assuming we don't hit any major blockers.
